Caused by: java.util.concurrent.RejectedExecutionException: Thread limit exceeded replacing blocked worker

See https://github.com/redhat-developer/lsp4ij/discussions/1558#discussioncomment-17172727

```
Caused by: java.util.concurrent.RejectedExecutionException: Thread limit exceeded replacing blocked worker
	at java.base/java.util.concurrent.ForkJoinPool.tryCompensate(ForkJoinPool.java:2223)
	at java.base/java.util.concurrent.ForkJoinPool.compensatedBlock(ForkJoinPool.java:4324)
	at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:4308)
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2116)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.blockUntilWriteActionIsDone(NonBlockingReadActionImpl.java:715)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$executeSynchronously$0(NonBlockingReadActionImpl.java:659)
	at com.intellij.openapi.application.impl.NbraUtilKt.runSynchronousNonBlockingReadActionWithCompensation$lambda$0(nbraUtil.kt:15)
	at kotlinx.coroutines.internal.intellij.IntellijCoroutines.runAndCompensateParallelism-VtjQ1oo(intellij.kt:91)
	at com.intellij.util.IntelliJCoroutinesFacade.runAndCompensateParallelism-VtjQ1oo(IntelliJCoroutinesFacade.kt:45)
	at com.intellij.openapi.application.impl.NbraUtilKt.runSynchronousNonBlockingReadActionWithCompensation(nbraUtil.kt:14)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.executeSynchronously(NonBlockingReadActionImpl.java:643)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl.executeSynchronously(NonBlockingReadActionImpl.java:295)
	at com.redhat.devtools.lsp4ij.internal.ApplicationUtils.runCancellableReadAction(ApplicationUtils.java:54)
	at com.redhat.devtools.lsp4ij.LSPIJUtils.getDocument(LSPIJUtils.java:604)
	at com.redhat.devtools.lsp4ij.LSPIJUtils.getDocument(LSPIJUtils.java:615)
	at com.redhat.devtools.lsp4ij.usages.LSPUsageSupport.lambda$collectUsages$2(LSPUsageSupport.java:83)
```

--- a/src/main/java/com/redhat/devtools/lsp4ij/LSPIJUtils.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/LSPIJUtils.java
@@ -611,8 +611,8 @@ public class LSPIJUtils {
      * @return the document corresponding to the PSI element, or {@code null} if no document could be found
      */
     public static @Nullable Document getDocument(@NotNull PsiElement element) {
-        VirtualFile virtualFile = getFile(element);
-        return virtualFile != null ? getDocument(virtualFile) : null;
+        PsiFile psiFile = element.getContainingFile();
+        return psiFile != null ? psiFile.getViewProvider().getDocument() : null;
     }
 
     /**
@@ -621,7 +621,7 @@ public class LSPIJUtils {
      * @param documentUri the uri of the Document to return
      * @return the @{@link Document} associated to <code>documentUri</code>, or <code>null</code>
      */
-    public static @Nullable Document getDocument(URI documentUri) {
+    public static @Nullable Document getDocument(@Nullable URI documentUri) {
         if (documentUri == null) {
             return null;
         }

